docs/architecture/whitelisting.md

```diff
@@ -75,3 +76,60 @@ the following procedure:
 After these steps are done, the ERC20 whitelisting is finished. Liquidity for
 the given asset should promptly begin deposits in order to provide a smooth
 experience for users interested in interacting with the token.
+
+## Detailed Procedure
+### Prerequisites for Submission
+
+Before initiating a governance proposal to whitelist an ERC-20 token as a ZRC-20, ensure you meet the following prerequisites:
+
+* **Zetacored CLI Installation**: Make sure the zetacored CLI tool is installed on your system. This tool is essential for interacting with the Zetachain network and submitting proposals.
+* **Wallet Setup and Funding**: A wallet must be set up and associated with the zetacored CLI. This wallet should contain a minimum of 1000 ZETA tokens, plus additional ZETA for gas fees, to cover the deposit required for proposal submission.
+
+### **Submitting a Governance Proposal**
+
+To submit a governance proposal for whitelisting an ERC-20 token, follow these steps:
+
+1.  **Prepare the Proposal JSON File**: Draft a `mainnet_proposal.json` file that outlines the details of the token you wish to whitelist. This file must include:
+    
+    -   The proposal title and description, clearly articulating the value and details of the token and how it benefits the Zetachain ecosystem.
+    -   A deposit amount in ZETA to accompany the proposal. Mainnet minimum is currently `1000 Zeta` or `1000000000000000000000azeta`
+    
+    Example `mainnet_proposal.json`:
+```json
+{ "messages": 
+	[ { "@type": "/cosmos.gov.v1.MsgExecLegacyContent", 
+		"content": { 
+					"@type": "/cosmos.gov.v1beta1.TextProposal", 
+					 "title": "Whitelist Token XYZ from XYZ Protocol", 
+					 "description": "The current address is 0xADDRESSONMAINNET on ETH and is fully verified 			https://etherscan.io/token/ADDRESSONMAINNET#code. Whitelisting will allow us to launch our dApp fully and start onboarding users to Zetachain. XYZ's core product is a key component of the Zetachain ecosystem that is yet to be deployed. Whitelisting would mean we could rapidly deploy this." 
+				}, 
+		"authority": "zeta10d07y265gmmuvt4z0w9aw880jnsr700jvxasvr" 
+	} ], 
+	"metadata": "ipfs://YOUR_IPFS_METADATA.json", 
+	"deposit": "1000000000000000000000azeta" 
+}
+```
+2. **Prepare the IPFS Metadata JSON**: Create a metadata file to be uploaded to IPFS. This file should provide additional context and details about the proposal, including the title, authors, summary, detailed description, and a link to a forum or Discord for community discussion.
+
+Example IPFS Metadata JSON:
```

Comment:
It's a good idea to add a template `metadata.json`, I propose we add it to the following section:

https://github.com/zeta-chain/docs/blob/6c9c7aba02963825b332fb76f6ba6a9b076797b2/docs/developers/cli/governance/overview.mdx#L19

Reply:
This CLI command produces an incorrect JSON file, and I intend to fix it as well once these doc changes get merged.

Reply:
Which CLI command? `zetacored tx gov draft-proposal`?

Reply:
Yes, the output is not correct for this.